Cargo.toml

```diff
@@ -336,6 +338,7 @@ nexus-test-utils = { path = "nexus/test-utils" }
 nexus-types = { path = "nexus/types" }
 num-integer = "0.1.46"
 num = { version = "0.4.2", default-features = false, features = [ "libm" ] }
+num_cpus = "1.16.0"
```

Comment:
This could use https://doc.rust-lang.org/std/thread/fn.available_parallelism.html instead.